portfolio/src/main/webapp/script.js

```diff
@@ -15,30 +20,163 @@ function plusSlides(nextSlide, slideSection) {
 // nextSlide represents whether the slide will switch to the next or previous slide
 // slideSection represents which section of slides is receving the change as referenced by "slideSections"
 function showSlides(nextSlide, slideSection) {
-  var i;
-  var slides = document.getElementsByClassName(slideSections[slideSection]);
-  if (nextSlide> slides.length) {slideIndices[slideSection] = 1}    
-  if (nextSlide < 1) {slideIndices[slideSection] = slides.length}
-  for (i = 0; i < slides.length; i++) {
-      slides[i].style.display = "none";  
-  }
-    slides[slideIndices[slideSection]-1].style.display = "block";  
-}
-
-function getComments() {
-   fetch('/data').then(response => response.json()).then((commentInfo) => {
-    const commentBox = document.getElementById("comment-box");
-    const numComments = commentInfo.numComments;
-    const comments = commentInfo.commentList;
+    var i;
+    var slides = document.getElementsByClassName(slideSections[slideSection]);
+
+    if (slideSection > 0) {
+        modals = document.getElementsByClassName(modalBoxes[slideSection - 1]);
+    }
+
+    if (nextSlide > slides.length) {
+        slideIndices[slideSection] = 1;
+    }
+    if (nextSlide < 1) {
+        slideIndices[slideSection] = slides.length;  
+    }
+
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";
 
-    for (i = 0; i < numComments; i++) {
-        commentBox.appendChild(createComment(comments[i]));
+        if (slideSection > 0) {
+            modals[i].style.display = "none";
+        }
+
+        if (slideSection == 0) {
```

Comment:
Can SlideSection be negative?  Otherwise 'else' will work.

Reply:
I suggest adding a brief comment and structuring the statement as:

if (slideSection == 0) {
...
} else {
...
}

This makes it clear that you are doing something different for value 0.

Reply:
I added in comments and included the else body.